File: app/src/main/java/ru/nikshlykov/englishwordsapp/Mode.java
package ru.nikshlykov.englishwordsapp;

import androidx.annotation.NonNull;
import androidx.room.ColumnInfo;
import androidx.room.Entity;
import androidx.room.PrimaryKey;

@Entity(tableName = "Modes")
public class Mode {

    @PrimaryKey()
    @NonNull
    @ColumnInfo(name = "_id")
    public long id;

    @NonNull
    @ColumnInfo(name = "ModeName")
    public String name;

    @NonNull
    @ColumnInfo(name = "IsSelected", defaultValue = "0")
    public int isSelected; // Параметр выбора режима изучения слов (1 - выбран, 0 - не выбран).

    @NonNull
    @ColumnInfo(name = "ImageResourceId")
    public String imageResourceId; // id картинки (скрин режима) для вывода в Activity режимов.
}

File: app/src/main/java/ru/nikshlykov/englishwordsapp/db/models/Mode.kt
package ru.nikshlykov.englishwordsapp.db.models

import androidx.room.ColumnInfo
import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "Modes")
class Mode {

  @PrimaryKey
  @ColumnInfo(name = "_id")
  var id: Long = 0

  @ColumnInfo(name = "ModeName")
  var name: String = ""

  @ColumnInfo(name = "IsSelected", defaultValue = "0")
  var selected: Int = 0 // Параметр выбора режима изучения слов (1 - выбран, 0 - не выбран).

  @ColumnInfo(name = "ImageResourceId")
  var imageResourceId: String = "" // id картинки (скрин режима) для вывода в Activity режимов.
}